compiler/test/suites/functions.re

```diff
@@ -385,4 +385,173 @@ truc()|},
     |},
     "which has a default value, but the matching argument does not.",
   );
+
+  assertRun(
```

Comment:
Should we have a test where a module provides a partially applied function, also maybe a clear test where a function returns a partially applied function. something like:
```
let f = (x, y) => {
  let t  = (a, b) => x + a + b
  let y = partial (_, y)
  y
}
print(f(1, 2)(3)) // 6
```
this ensures that we don't break closures and ensures that we are not losing the partial information when returning from a scope (The way this is implemented I don't think those are major concerns but they seem like good sanity checks. I think your test with the pipeline operator somewhat tests this but doesn't test closures.